kitchen/bin/run-proxy-kitchen.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Usage: run-proxy-kitchen.sh http
+# Usage: run-proxy-kitchen.sh h2c
+# Usage: run-proxy-kitchen.sh http /usr/local/bin/haproxy
+#
+# Launches kitchen fronted by a HAProxy instance.
+
+set -eux
+echo "run-proxy-kitchen.sh: starting..."
+
+KITCHEN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )"
+echo "run-proxy-kitchen.sh: kitchen directory is ${KITCHEN_DIR}"
+
+PROTO_VERSION=${1:-http1}
+echo "run-proxy-kitchen.sh: requested to launch kitchen with protocol ${PROTO_VERSION}"
+
+PROXY_CMD=$(which nginx)
+echo "run-proxy-kitchen.sh: nginx proxy command is ${PROXY_CMD}"
+
+cd ${KITCHEN_DIR}
+
+echo "run-proxy-kitchen.sh: launching kitchen on port ${PORT1}"
+${KITCHEN_DIR}/bin/kitchen -p ${PORT1} &
+
+echo "run-proxy-kitchen.sh: generating proxy config file at ${MESOS_SANDBOX}/nginx-${PROTO_VERSION}-server.conf"
+envsubst '${PORT0} ${PORT1}' < "${KITCHEN_DIR}/bin/nginx-${PROTO_VERSION}-server.conf" > "${MESOS_SANDBOX}/nginx-${PROTO_VERSION}-server.conf"
```

Comment:
Unlike HAProxy, nginx configuration files do not directly support environment variable expansion.